File: python/resistor-color-trio/resistor_color_trio.py
# pull in the code from the previous exercises
from resistor_color     import color_code
from resistor_color_duo import value


def three_band_value(colors):
    return value(colors[:2]) * pow(10, color_code(colors[2]))


def value_label(value):
    idx = 0
    while value > 0 and value % 1000 == 0:
        value //= 1000
        idx += 1
    prefix = ["", "kilo", "mega", "giga"][idx]
    return f'{value} {prefix}ohms'


def label(colors):
    return value_label(three_band_value(colors))

File: php/resistor-color-trio/ResistorColorTrio.php
<?php

declare(strict_types=1);

require_once 'ResistorColor.php';
require_once 'ResistorColorDuo.php';

class ResistorColorTrio
{
    public function label(array $colors): string
    {
        $value = (new ResistorColorDuo())->getColorsValue(array_slice($colors, 0, 2));
        $value *= 10 ** colorCode($colors[2]);

        $idx = 0;
        if ($value > 0) {
            while ($value % 1000 == 0) {
                $value /= 1000;
                $idx += 1;
            }
        }
        $prefix = ['', 'kilo', 'mega', 'giga'][$idx];

        return "$value {$prefix}ohms";
    }
}
